### ISSUE TITLE
(node:533) UnhandledPromiseRejectionWarning: RangeError: Variable $1 out of range. Parameters array length: 0

### ISSUE THREAD

**Issue by IsaiahJTurner** [dmfay/massive-js#642]
## Summary
I need to run this SQL for AWS RDS to setup postgis but it fails with the following error. I've tried using scripts as well as db.query.
### Code demonstrating the behavior
```js
db.query(`CREATE FUNCTION exec(text) returns text language plpgsql volatile AS $f$ BEGIN EXECUTE $1; RETURN $1; END; $f$;`)
```
I also tried
```js
  const { fd, path, cleanup } = await file()
  fs.writeFileSync(fd, `
    CREATE FUNCTION exec(text) returns text language plpgsql volatile AS $f$ BEGIN EXECUTE $1; RETURN $1; END; $f$;
  `)
  const sql = new db.pgp.QueryFile(path, { minify: true })
  await db.query(sql)
  cleanup()
```

## Expected behavior
Creates the function

## Actual behavior
(node:533) UnhandledPromiseRejectionWarning: RangeError: Variable $1 out of range. Parameters array length: 0

**Comment by vitaly-t** [comment]
Your code examples and the error do not add up. The only way such an error would pop-up, if you were to pass in `[]` as the formatting parameter into the query method:

```js
await db.query(sql, [])
```

You should check your code again, and make sure you do not pass it an empty array, as it would try to format the query, and throw that error, and correctly throw that error.

**Comment by vitaly-t** [comment]
@dmfay There seems to be major problem with your implementation of method `query`:

```js
Database.prototype.query = function (query, params = [], options = {}) {
```

You cannot default formatting parameters to `[]`, you have to default it to `undefined`. That's how `pg-promise` works. If it finds `[]`, it runs `format` on the query, for the syntax of `$1, $2, ...`, if it finds any such variable, it replaces those. And because you default it incorrectly, any query with text that looks like those variables will result in that `RangeError`, as reported.

I'm a bit surprised this didn't come up earlier.

**Comment by dmfay** [comment]
Fixed in 5.5.1.
